ui/ui.go

```diff
@@ -0,0 +1,16 @@
+package ui
+
+import "fmt"
+
+func UI() (string, error) {
+	var input string
+
+	fmt.Printf("Type 'help' for list of commands\n>")
+
+	_, iErr := fmt.Scan(&input)
+	if iErr != nil {
+		return input, iErr
```

Comment:
very common error handling pattern:

```suggestion
	if _, err := fmt.Scan(&input); err != nil {
		return input, err
```

Reply:
but also, in this case, you don't have to check if the error is nil, you could do

```go
  ...
	_, iErr := fmt.Scan(&input)
	return input, iErr
}
```

that will make the same function

Reply:
Makes sense, done!